moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -409,18 +412,30 @@ bool IKFastKinematicsPlugin::initialize(const moveit::core::RobotModel& robot_mo
   }
 
   storeValues(robot_model, group_name, base_frame, tip_frames, search_discretization);
+  if (!lookupParam("linkprefix", linkframeprefix_, std::string("")))
+  {
+    ROS_INFO_NAMED(name_, "linkprefix parameter not set. Assuming empty prefix.");
+  }
+  else
+  {
+    ROS_DEBUG_STREAM_NAMED(name_, "using linkprefix " << linkframeprefix_);
+  }
 
   // This IKFast solution was generated with IKFAST_TIP_FRAME_ and IKFAST_BASE_FRAME_.
   // It is often the case that fixed joints are added to these links to model things like
   // a robot mounted on a table or a robot with an end effector attached to the last link.
   // To support these use cases, we store the transform from the IKFAST_BASE_FRAME_ to the
   // base_frame_ and IKFAST_TIP_FRAME_ the tip_frame_ and transform to the input pose accordingly
-  if (!computeRelativeTransform(tip_frames_[0], IKFAST_TIP_FRAME_, group_tip_to_chain_tip_, tip_transform_required_) ||
-      !computeRelativeTransform(IKFAST_BASE_FRAME_, base_frame_, chain_base_to_group_base_, base_transform_required_))
+  if (!computeRelativeTransform(tip_frames_[0], linkframeprefix_ + IKFAST_TIP_FRAME_, group_tip_to_chain_tip_,
```

Comment:
Will the function  `computeRelativeTransform` print an error when the combined link prefix and hardcoded IKFAST frame names do not exists?  If not then we should check if the resulting link name is valid

Reply:
computeRelativeTransform printed a duplicated warning with the frame name; `getLinkModel()` spits an error as well. But to draw the user's attention to the `link_prefix` parameter I've added explicit checks with appropriate error messages.